test_main_objs.py

```diff
@@ -0,0 +1,228 @@
+    'https://crimproject.org/mei/CRIM_Mass_0018_1.mei',
+    'https://crimproject.org/mei/CRIM_Mass_0018_2.mei',
+    'https://crimproject.org/mei/CRIM_Mass_0018_3.mei',
+    'https://crimproject.org/mei/CRIM_Mass_0018_4.mei',
+    'https://crimproject.org/mei/CRIM_Mass_0018_5.mei',
+    'https://crimproject.org/mei/CRIM_Mass_0019_1.mei',
+    'https://crimproject.org/mei/CRIM_Mass_0019_2.mei',
+    'https://crimproject.org/mei/CRIM_Mass_0019_3.mei',
+    'https://crimproject.org/mei/CRIM_Mass_0019_4.mei',
+    'https://crimproject.org/mei/CRIM_Mass_0019_5.mei',
+    'https://crimproject.org/mei/CRIM_Mass_0020_1.mei',
+    'https://crimproject.org/mei/CRIM_Mass_0020_2.mei',
+    'https://crimproject.org/mei/CRIM_Mass_0020_3.mei',
+    'https://crimproject.org/mei/CRIM_Mass_0020_4.mei',
+    'https://crimproject.org/mei/CRIM_Mass_0020_5.mei',
+    'https://crimproject.org/mei/CRIM_Mass_0021_1.mei',
+    'https://crimproject.org/mei/CRIM_Mass_0021_2.mei',
+    'https://crimproject.org/mei/CRIM_Mass_0021_3.mei',
+    'https://crimproject.org/mei/CRIM_Mass_0021_4.mei',
+    'https://crimproject.org/mei/CRIM_Mass_0021_5.mei',
+    'https://crimproject.org/mei/CRIM_Mass_0022_1.mei',
+    'https://crimproject.org/mei/CRIM_Mass_0022_2.mei',
+    'https://crimproject.org/mei/CRIM_Mass_0022_3.mei',
+    'https://crimproject.org/mei/CRIM_Mass_0022_4.mei',
+    'https://crimproject.org/mei/CRIM_Mass_0022_5.mei'
+]
+
+def build_crim_models(files):
+    """
+    Build a list of crim models for the selected files.
+    """
+    root = "https://raw.githubusercontent.com/CRIM-Project/CRIM-online/master/crim/static/mei/MEI_3.0/"
+    prefix = "CRIM_Model_00"
+    postfix = ".mei"
+    models = {}
+    for file in files:
+        try:
+            if not "http" in file:
+                corpus = CorpusBase([root + prefix + file + postfix])
+            else:
+                file = file.split("/")[-1]
+                corpus = CorpusBase([root + file])
+            model = corpus.scores[0]
+            models[file] = model
+        except:
+            print("Can't import file", file)
+
+    return models
+
+def validate_ngrams_last_offsets(model, df, n, how='columnwise', other=None, held='Held',
+                                 exclude=['Rest'], interval_settings=('d', True, True), unit=0):
+
+    df1 = model.getNgrams(df=df, n=n, how=how, other=other, held=held,
+                  exclude=exclude, interval_settings=interval_settings, unit=unit,
+                  offsets='first')
+
+    df2 = model.getNgrams(df=df, n=n, how=how, other=other, held=held,
+                  exclude=exclude, interval_settings=interval_settings, unit=unit,
+                  offsets='last')
+
+    # compare patterns
+    df1_cols = [df1.iloc[:, i] for i in range(len(df1.columns))]
+    df2_cols = [df1.iloc[:, i] for i in range(len(df2.columns))]
+
+    # making sure that we would get the same pattern whichever offsets we choose.
```

Comment:
It's good practice to put the objective of the test in a docstring of the function.